src/pages/iou/request/step/IOURequestStepDate.js

```diff
@@ -0,0 +1,86 @@
+import dateAdd from 'date-fns/add';
+import dateSubtract from 'date-fns/sub';
+import React from 'react';
+import DatePicker from '@components/DatePicker';
+import FormProvider from '@components/Form/FormProvider';
+import transactionPropTypes from '@components/transactionPropTypes';
+import useLocalize from '@hooks/useLocalize';
+import compose from '@libs/compose';
+import * as IOUUtils from '@libs/IOUUtils';
+import Navigation from '@libs/Navigation/Navigation';
+import useThemeStyles from '@styles/useThemeStyles';
+import * as IOU from '@userActions/IOU';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import IOURequestStepRoutePropTypes from './IOURequestStepRoutePropTypes';
+import StepScreenWrapper from './StepScreenWrapper';
+import withFullTransactionOrNotFound from './withFullTransactionOrNotFound';
+import withWritableReportOrNotFound from './withWritableReportOrNotFound';
+
+const propTypes = {
+    /** Navigation route context info provided by react navigation */
+    route: IOURequestStepRoutePropTypes.isRequired,
+
+    /** Onyx Props */
+    /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
+    transaction: transactionPropTypes,
+};
+
+const defaultProps = {
+    transaction: {},
+};
+
+function IOURequestStepDate({
+    route: {
+        params: {iouType, backTo, transactionID},
+    },
+    transaction,
+}) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    const navigateBack = () => {
+        Navigation.goBack(backTo || ROUTES.HOME);
+    };
+
+    /**
+     * @param {Object} value
+     * @param {String} value.moneyRequestCreated
+     */
+    const updateDate = (value) => {
+        IOU.setMoneyRequestCreated_temporaryForRefactor(transactionID, value.moneyRequestCreated);
+        navigateBack();
+    };
+
+    return (
+        <StepScreenWrapper
+            headerTitle={translate('common.date')}
+            onBackButtonPress={navigateBack}
+            shouldShowNotFound={!IOUUtils.isValidMoneyRequestType(iouType)}
+            shouldShowWrapper
+            testID={IOURequestStepDate.displayName}
+        >
+            <FormProvider
+                style={[styles.flexGrow1, styles.ph5]}
+                formID={ONYXKEYS.FORMS.MONEY_REQUEST_DATE_FORM}
+                onSubmit={updateDate}
+                submitButtonText={translate('common.save')}
+                enabledWhenOffline
+            >
+                <DatePicker
+                    inputID="moneyRequestCreated"
+                    label={translate('common.date')}
+                    defaultValue={transaction.created}
+                    maxDate={dateAdd(new Date(), {years: 1})}
```

Comment:
@tgolen Hi, I come from https://github.com/Expensify/App/issues/33013 Do we allow setting the date to the future date ?

Reply:
Yes, we changed the behavior to allow that

Reply:
Yes, it was an intentional change.

Reply:
This change caused https://github.com/Expensify/App/issues/33013 where when editing a request you couldn't change the date if it was a future date. We needed to update `EditRequestCreatedPage` to reflect the updated `maxDate` too.